inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -638,7 +638,22 @@ void ReversePermutationsPass::run() {
 
 void RemovePermutationsNHWCToNCHWPass::run() {
     std::list<CNNLayerPtr> permutationsToRemove;
-
+    auto isPermuteToRemove = [](CNNLayerPtr layer, bool before) {
+        if (LayerInfo(layer).isPermute()) {
+            auto permute_order = layer->GetParamAsInts("order");
+            return before && permute_order == std::vector<int>({0, 3, 1, 2}) ||
+                   !before && permute_order == std::vector<int>({0, 2, 3, 1});
+        }
+        if (layer->outData.size() != 1) return false;
+        if (getInputTo(layer->outData[0]).size() != 1) return false;
+        if (!LayerInfo(layer).isReshape()) return false;
```

Comment:
i would not remove reshape layers at all - pass called remove permutations isnt it?

Reply:
Reshapes are not actually removed in this pass, only an order of their output dimensions can be changed.

Reply:
why not - this lambda accept layer ptr, and at line 649 i see that permute layer type get passed further. For example i'll call this lambda with reshape layer before and reshape layer after and get true as the result. And what is the point ?